### PR TITLE
Make gp_acquisition_func public

### PR DESCRIPTION
**Pull request by MechCoder**
A minor pull request to make `_acquistion_func` public. Also renamed it as `gp_acquisition_func` since the actual acquisition function depends on the distribution of the posterior

### CODE REVIEW

File: skopt/gp_opt.py [hunk 1]
```diff
@@ -8,16 +8,53 @@
...
+    ----------
+    x0 : array-like
+        Values where the acquistion function should be computed.
+
+    gp: GaussianProcess model
+        The fit sklearn gaussian process estimator that approximates
+        the function.
+
+    prev_best: float, optional
+        The previousbest value over which we want to improve.
```
Comment (betatim) [inline skopt/gp_opt.py]: -> previous best

**Comment by betatim** [comment]
Does it need to be `gp_...` it would be useful also for a tree based model instead of a GP. As long as both provide a central value and uncertainty.

**Comment by MechCoder** [comment]
Addressed. Please merge if happy.

**Comment by MechCoder** [comment]
@betatim Can I merge this? Sorry for rushing, but I'm refactoring some more code..

**Pull request merged**